src/main/java/com/assignment/application/service/CachingInfo.java

```diff
@@ -25,51 +29,87 @@
-        List<CompleteCompInfo> companyInfoList = companyRepo.getCompanyCompleteInfo(companyId);
-        return companyInfoList;
+    //@Cacheable(value = "companyCompleteInfo", key = "#companyId", condition = "#result==null")
+    public Page<CompleteCompInfo> getCompanyCompleteInfo(Long companyId, Pageable pageable) {
+        return companyRepo.getCompanyCompleteInfo(companyId, pageable);
     }
 
-    @Cacheable(value = "companyEmployeeList", key = "#companyId", condition = "#result==null")
+    //@Cacheable(value = "companyEmployeeList", key = "#companyId", condition = "#result==null")
     public List<Employee> getEmployeeOfComp(Long companyId) {
-        List<Employee> employeesList = employeeRepo.getAllEmpByCompId(companyId);
-        return employeesList;
+        return employeeRepo.getAllEmpByCompId(companyId);
     }
 
-    @Caching(evict = {@CacheEvict(value = "companyEmployeeList", key = "#companyId"), @CacheEvict(value = "companyCompleteInfo", key = "#companyId")})
+    //@Caching(evict = {@CacheEvict(value = "companyEmployeeList", key = "#companyId"), @CacheEvict(value = "companyCompleteInfo", key = "#companyId")})
     public String updateEmployeeInfo(String employeeId, Long companyId, EmployeeInfoUpdate employeeInfoUpdate) {
         Employee employee = employeeRepo.getEmployee(employeeId);
-        if (employee == null || !employee.getCompanyId().equals(companyId)) {
-            return StringConstant.INVALID_CREDENTIALS;
+        if (employee == null || employee.getIsActive() == 0) {
+            throw new NotExistsException("No such employee exists");
         }
-        if (!employeeInfoUpdate.getCurrentAddress().isEmpty()) {
-            employee.setCurrentAdd(employeeInfoUpdate.getCurrentAddress());
+        Department department = departmentRepo.findById(employee.getDepartmentId()).orElse(null);
+        if (department == null || department.getIsActive() == 0) {
+            throw new NotExistsException("No such department exists");
         }
-        if (!employeeInfoUpdate.getPermanentAddress().isEmpty()) {
-            employee.setPermanentAdd(employeeInfoUpdate.getPermanentAddress());
+        if (!department.getCompanyId().equals(companyId)) {
+            throw new DataMismatchException("Company Id is not valid for the given employee");
         }
-        if (!employeeInfoUpdate.getPosition().isEmpty()) {
-            employee.setPosition(employeeInfoUpdate.getPosition());
+        if (!employeeInfoUpdate.getCurrentAddress().isEmpty()) {
+            employee.setCurrentAddress(employeeInfoUpdate.getCurrentAddress());
         }
-        if (!employeeInfoUpdate.getPhoneNumber().isEmpty()) {
-            employee.setPhoneNumber(employeeInfoUpdate.getPhoneNumber());
+        if (!employeeInfoUpdate.getPermanentAddress().isEmpty()) {
+            employee.setPermanentAddress(employeeInfoUpdate.getPermanentAddress());
         }
+        employee.setUpdatedAt(new Date());
```

Comment:
no need to set UpdatedAt manually. declare a annotation on updated_at which will set the date